### PR TITLE
article upload feature

### PR DESCRIPTION
**Pull request by mboyea**
Created a CLI tool to manage articles `pnpm run article` (stored in scripts/article.sh)
Implemented the help screen, upload feature, and print feature of the article script (upload feature includes loading .env variables and opening a proxy into the database)
Implemented full range of upload options, allowing the user to pass filenames or raw text
Wrote readme documentation to support use of the article script

changed `pnpm run dev` command to `pnpm run start` for better cohesiveness with other UI framework environments
changed `pnpm run preview` command to `pnpm run start:preview` for better cohesiveness with other script naming conventions
fixed the cancel behavior of pnpm run start scripts (it properly closes the proxy into the database)

### CODE REVIEW

**Comment by mboyea** [comment]
**Screenshot**
image

File: scripts/article.sh [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  title,
+  url
+) VALUES (
+$([[ -n "$description" ]] && echo "  E'$(echo "$description" | sed -e s/\'/\\\\\'/g)',")
+$([[ -n "$summary" ]] && echo "  E'$(echo "$summary" | sed -e s/\'/\\\\\'/g)',")
+  E'$(echo "$text" | sed -e s/\'/\\\\\'/g)',
+  '$(echo "$title" | sed -e s/\'/\\\\\'/g)',
+  '$(echo "$url" | sed -e s/\'/\\\\\'/g)'
+);
+EOF
```
Comment (dbooyah93) [inline scripts/article.sh:77]: The indentation here seems wrong

File: scripts/article.sh [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+create_proxy() {
+	# prepare proxy program cleanup
+	cleanup() {
+		# kill proxy
+		kill -INT "$proxy_program_PID"
+		sleep 0.4
+	}
+	trap cleanup EXIT
+	# create proxy to connect to database on port 5432
```
Comment (dbooyah93) [inline scripts/article.sh:43]: Further below in the deploy-database file, you put spaces between your comments. I do suggest adding extra space where it makes sense as long as your compiler removes blank spaces.

File: scripts/deploy-database.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+# create proxy to connect to database on port 5432
+coproc proxy_program (flyctl proxy 5432 -a mboyea-database-test)
+
+# load .env
+set -a; source .env; set +a
+
+# wait for output from proxy program
+read output <&"${proxy_program[0]}"
+# print output from proxy program
```
Comment (dbooyah93) [inline scripts/deploy-database.sh]: There's more of the same inconsistent spacing when it comes to comments. I prefer to add the extra spaces all the time; consider doing the same

**Review by dbooyah93 — APPROVED** [review]
Try to keep commit messages to 72 characters so your messages don't get paginated with ellipses in the commit tab.

File: scripts/article.sh (hunk 1 above)
Comment (mboyea) [reply scripts/article.sh:77]: Ah, the range between EOF ... EOF is equivalent to an inline string in Bash.
Any extraneous tabs would be included in the SQL code passed to psql, which is not desirable.
So here, the reduced indentation is intentional.
Also, the EOF indicator would not be recognized if a [TAB] or a [SPACE] were before it, so indentation is not supported in this context - the above approach is correct.

File: scripts/deploy-database.sh (hunk 3 above)
Comment (mboyea) [reply scripts/deploy-database.sh]: Good points about inconsistent spacing

**Comment by mboyea** [comment]
> Try to keep commit messages to 72 characters so your messages don't get paginated with ellipses in the commit tab.

Fair enough, these could be more concise.

**Pull request merged**